test/index.test.ts

```diff
@@ -56,6 +55,42 @@ describe("Anchor", () => {
     expect(typeof token).toBe("string");
   });
 
+  it("should be able to authenticate with client domain", async () => {
+    const auth = await anchor.auth();
+    let signedByClient = false;
+    let signedByDomain = false;
+
+    const walletSigner = {
+      signWithClientAccount: (txn, account) => {
+        txn.sign(account);
+        signedByClient = true;
+        return txn;
+      },
+      signWithDomainAccount: (transactionXDR, networkPassPhrase, account) => {
+        // dummy secret key for signing
+        const clientDomainKp = Keypair.fromSecret(
+          "SC7PKBRGRI5X4XP4QICBZ2NL67VUJJVKFKXDTGSPI3SQYZGC4NZWONIH"
+        );
+        const transaction = StellarSdk.TransactionBuilder.fromXDR(
+          transactionXDR,
+          networkPassPhrase
+        );
+        transaction.sign(clientDomainKp);
+        signedByDomain = true;
+        return transaction;
+      },
+    };
+
+    const challengeResponse = await auth.challenge(
```

Comment:
Can you just call auth.authenticate and check that token has been returned?

Reply:
it's because the "demo-wallet-server.stellar.org" domain secret key I don't have (just using a random secret key up at line 71), so it would fail to authenticate. So I'm testing up to that point.

But I don't think we should add it to the test though just to be safe? wdyt?

Reply:
added a comment there to make more clear